### ISSUE TITLE
No app capable of handling camera intent for android11

### ISSUE THREAD

**Issue by bharathgrandhe-utu** [jkwiecien/EasyImage#230]
Getting the above error in android11

**Comment by firecatmusic** [comment]
same issue

**Comment by hafiz013** [comment]
so what news need to be update?

**Comment by RkNaing** [comment]
May be adding the queries & use-feature tag in the AndroidManifest.xml help.....

`
<uses-feature
        android:name="android.hardware.camera.any"
        android:required="false" />

    <uses-feature
        android:name="android.hardware.camera.autofocus"
        android:required="false" />

<queries>
        <intent>
            <action android:name="android.media.action.IMAGE_CAPTURE" />
        </intent>
        <intent>
            <action android:name="android.intent.action.PICK" />
        </intent>
    </queries>

`

**Comment by markizdeviler** [comment]
adding queries working for me.
https://developer.android.com/training/camera/photobasics#TaskCaptureIntent

**Comment by bpappin** [comment]
screenshots are terrible documentation!

**Comment by njamal816** [comment]
For Google Search and get easier results for all.


    <intent>
        <action android:name="android.media.action.IMAGE_CAPTURE" />
    </intent>
    
    <intent>
        <action android:name="android.intent.action.GET_CONTENT" />
        <data android:mimeType="image/*" />
    </intent>
    <intent>
        <action android:name="android.intent.action.PICK" />
        <data android:mimeType="image/*" />
    </intent>
    <intent>
        <action android:name="android.intent.action.CHOOSER" />
     </intent>
</queries>

**Comment by Khenisiddharth** [comment]
I have the same issue. Is there any proper solution?

**Comment by njamal816** [comment]
@Khenisiddharth Have you added the above queries?

**Comment by Khenisiddharth** [comment]
Yes, I have added the above queries. Here is my code it's working fine in another android version. 
I have the only issue in android 11.

String destination = Environment.getExternalStorageDirectory().getPath() + "/image.jpg";
                imageUri = Uri.fromFile(new File(destination));
                Intent intent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
                intent.putExtra(MediaStore.EXTRA_OUTPUT, imageUri);
                startActivityForResult(intent, Constant.REQUEST_CAMERA);

**Comment by njamal816** [comment]
@Khenisiddharth  The issue might be with the destination as you are not allowed to access root of External storage directly. You may try accessing the Document, image or download location. You can try mediastore as well.

**Comment by Khenisiddharth** [comment]
Is there any demo code or example? I have no idea how to use mediastore in intent.

**Comment by njamal816** [comment]
Instead of 
`Environment.getExternalStorageDirectory().getPath() + "/image.jpg"`
Try
`Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOCUMENTS).getAbsolutePath() + File.separator + yourappnamestring + "/image.jpg"`

**Comment by Khenisiddharth** [comment]
Result code 0 found.

  @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);

          Logger.print("<<<<<< print result code : "+resultCode);

       if (resultCode == RESULT_OK &&  requestCode == Constant.REQUEST_CAMERA){

            
       }
}
---------------------------------OUT PUT ----------------------------------

<<<<<< print result code : 0
